docs/root/version_history/current.rst

```diff
@@ -89,6 +89,7 @@ New Features
 * hds: added :ref:`cluster_endpoints_health <envoy_v3_api_field_service.health.v3.EndpointHealthResponse.cluster_endpoints_health>` to HDS responses, keeping endpoints in the same groupings as they were configured in the HDS specifier by cluster and locality instead of as a flat list.
 * hds: added :ref:`transport_socket_matches <envoy_v3_api_field_service.health.v3.ClusterHealthCheck.transport_socket_matches>` to HDS cluster health check specifier, so the existing match filter :ref:`transport_socket_match_criteria <envoy_v3_api_field_config.core.v3.HealthCheck.transport_socket_match_criteria>` in the repeated field :ref:`health_checks <envoy_v3_api_field_service.health.v3.ClusterHealthCheck.health_checks>` has context to match against. This unblocks support for health checks over HTTPS and HTTP/2.
 * hot restart: added :option:`--socket-path` and :option:`--socket-mode` to configure UDS path in the filesystem and set permission to it.
+* http: added HTTP/2 support for :ref:`connection keepalive <envoy_v3_api_field_config.core.v3.Http2ProtocolOptions.connection_keepalive>` via PING.
```

Comment:
Re: how to hook this into documentation to folks know to use it. I suggest that we update all examples that use xDS in the tree to use keepalive. Also, we can provide guidance in https://www.envoyproxy.io/docs/envoy/latest/configuration/overview/xds_api and https://www.envoyproxy.io/docs/envoy/latest/api-v3/config/core/v3/grpc_service.proto#envoy-v3-api-msg-config-core-v3-grpcservice-envoygrpc.

We could also add a warning potentially when Envoy gRPC is used for xDS without this option enabled.

Reply:
Added docs.

Reply:
I didn't add any to `grpc_service` because that applies to things other than xds, so it didn't seem like the right place for this.

Reply:
I agree this is more general, but I think that when you do opt to use Envoy gRPC, you should set this as best practice, hence why I think it's good to provide a recommendation in those docs.

Reply:
Actually, ignore my last comment, I see where you've added the best practices, that's all good.